server/src/tracing.ts

```diff
@@ -0,0 +1,142 @@
+/**
+ * Tools for instrumenting code with traces. This is mainly designed to support
+ * Sentry's tracing features, but could be expanded support other approaches,
+ * like OpenTelemetry.
+ */
+
+import * as Sentry from "@sentry/node";
+import type { Span } from "@sentry/tracing";
+import type { SpanContext } from "@sentry/types";
+export type { Span } from "@sentry/tracing";
+
+interface SpanOptions extends SpanContext {
+  parentSpan?: Span;
+}
+
+/**
+ * Start a tracing span. The returned span should be explicitly ended with
+ * `finishSpan`. The created span will be a child of whatever span is currently
+ * active (and then become the current span itself), or if there is no current
+ * span or transaction, this will start one for you.
+ *
+ * Alternatively, can explicitly pass an actual span object to be the parent:
+ * `startSpan({ parentSpan: yourSpan })`. In this case, the new span won't
+ * automatically become the global "current" span.
+ */
+export function startSpan(options: SpanOptions): Span {
+  let { parentSpan, ...spanOptions } = options;
+
+  let scope;
+  if (!parentSpan) {
+    scope = Sentry.getCurrentHub().getScope();
+    parentSpan = scope.getSpan();
+  }
+
+  let newSpan: Span;
+  if (parentSpan) {
+    newSpan = parentSpan.startChild(spanOptions);
+  } else {
+    newSpan = Sentry.startTransaction(spanOptions as any);
+  }
```

Comment:
Ugh, just realized that, because `startTransaction` is an “extension” method (i.e. it doesn’t come from sentry/core or sentry/node), it doesn’t exist if `Sentry.init()` hasn’t been called, which is the case in some tests. That causes this to silently return `undefined` in those tests, which is causing them to fail. ugh.